core/src/main/java/org/apache/calcite/plan/AbstractRelOptPlanner.java

```diff
@@ -130,24 +125,15 @@ public void checkCancel() {
    *
    * @param rule Rule
    */
-  protected void mapRuleDescription(RelOptRule rule) {
-    // Check that there isn't a rule with the same description,
-    // also validating description string.
-
+  protected boolean mapRuleDescription(RelOptRule rule) {
+    // Check that there isn't a rule with the same description
     final String description = rule.toString();
     assert description != null;
-    assert !description.contains("$")
-        : "Rule's description should not contain '$': "
-        + description;
-    assert !INTEGER_PATTERN.matcher(description).matches()
-        : "Rule's description should not be an integer: "
```

Comment:
How should we validate the rule description pattern now ?

Reply:
When creating the rule